lib/src/Base/Optim/OptimizationResult.cxx

```diff
@@ -433,6 +430,74 @@ Graph OptimizationResult::drawOptimalValueHistory() const
   return result;
 }
 
+/* Computes the Lagrange multipliers associated with the constraints as a post-processing of the optimal point. */
+/* L(x, l_eq, l_lower_bound, l_upper_bound, l_ineq) = J(x) + l_eq * C_eq(x) + l_lower_bound * (x-lb)^+ + l_upper_bound * (ub-x)^+ + l_ineq * C_ineq^+(x)
+   d/dx(L = d/dx(J) + l_eq * d/dx(C_eq) + l_lower_bound * d/dx(x-lb)^+ + l_upper_bound * d/dx(ub - x)^+ + l_ineq * d/dx(C_ineq^+)
+
+   The Lagrange multipliers are stored as [l_eq, l_lower_bounds, l_upper_bounds, l_ineq], where:
+   * l_eq is of dimension 0 if no equality constraint, else of dimension the number of scalar equality constraints
+   * l_lower_bounds and l_upper_bounds are of dimension 0 if no bound constraint, else of dimension dim(x) for both of them
+   * l_ineq is of dimension 0 if no inequality constraint, else of dimension the number of scalar inequality constraints
+
+   so if there is no constraint of any kind, the Lagrange multipliers are of dimension 0.
+ */
+Point OptimizationResult::computeLagrangeMultipliers(const Point & x) const
+{
+  const Scalar maximumConstraintError = ResourceMap::GetAsScalar("OptimizationAlgorithm-DefaultMaximumConstraintError");
```

Comment:
It could even be an argument of the method, with the ResourceMap value as a default value

Reply:
oh, this one is private, I kept getLagrangeMultipliers which calls computeLagrangeMultipliers on x^

Reply:
@regislebrun an update on this: I removed the OptimizationResult attribute and deprecated OptimizationResult::getLagrangeMultipliers, OptimizationAlgo::computeLagrangeMultipliers, now we have only OptimizationResult::computeLagrangeMultipliers(x) and OptimizationResult::computeLagrangeMultipliers()

Reply:
The idea of having both getLagrangeMultipliers() and computeLagrangeMultipliers() was to avoid computing them twice if the user asks for them twice, in the spirit of Distribution::computeMean() and Distribution::getMean(), but I agree with you: it is overkill here.